test/org/zaproxy/zap/authentication/UsernamePasswordAuthenticationCredentialsUnitTest.java

```diff
@@ -0,0 +1,114 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.authentication;
+
+import net.sf.json.JSON;
+import org.junit.Before;
+import org.junit.Test;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.equalToIgnoringCase;
+
+import org.zaproxy.zap.extension.api.ApiResponse;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+
+/**
+ * @author Vahid Rafiei (@vahid_r)
+ */
+public class UsernamePasswordAuthenticationCredentialsUnitTest {
+
+    private UsernamePasswordAuthenticationCredentials usernamePasswordAuthenticationCredentials;
+    private UsernamePasswordAuthenticationCredentials notConfiguredInstance;
+    private String username = "myUsername";
+    private String password = "myPassword";
+
+    @Before
+    public void setUp() {
+        this.usernamePasswordAuthenticationCredentials = new UsernamePasswordAuthenticationCredentials(username, password);
+        this.notConfiguredInstance = new UsernamePasswordAuthenticationCredentials();
+    }
+
+    @Test
+    public void shouldBeConfiguredIfUsernameAndPasswordAreNotNull() throws Exception {
+        // Given/When
+        boolean isConfigured = usernamePasswordAuthenticationCredentials.isConfigured();
+
+        // Then
+        assertThat(isConfigured, is(true));
+    }
+
+    @Test
+    public void shouldNotBeConfiguredIfUsernameAndPasswordAreNull() throws Exception {
+        // Given/When
+        boolean isConfigured = notConfiguredInstance.isConfigured();
+
+        // Then
+        assertThat(isConfigured, is(false));
+    }
+
+
+    @Test
+    public void shouldEncodeMethodReturnNullPatternIfUsernameIsNull() throws Exception {
+        // Given
+        String nullPattern = "AA==";
+        username = null;
+        password = "something";
+        String stringSeparator = "|";
+        usernamePasswordAuthenticationCredentials = new UsernamePasswordAuthenticationCredentials(username, password);
+
+        // When
+        String encodedResult = usernamePasswordAuthenticationCredentials.encode(stringSeparator);
+
+        // Then
+        assertThat(encodedResult, is(nullPattern));
+    }
+
+    @Test
+    public void shouldEncodeUsernameAndPasswordWithTheCorrectFieldSeparator() throws Exception {
+        // Given
+        List<String> someCorrectSeparators = Arrays.asList("-", "|", "/", "\\", "+");
+        String aCorrectSeparator = someCorrectSeparators.get(new Random().nextInt(someCorrectSeparators.size()));
```

Comment:
It could test all separators, something like:
```Java
for (String correctSeparator : someCorrectSeparators) {
    // When
    String encodedUsernamePassword = usernamePasswordAuthenticationCredentials.encode(correctSeparator);

    // Then
    assertThat(String.format("Failed to encode with '%s'", correctSeparator), encodedUsernamePassword, notNullValue());
    assertThat(
            String.format("Failed to properly encode with '%s'", correctSeparator),
            encodedUsernamePassword,
            is("bXlVc2VybmFtZQ==~bXlQYXNzd29yZA==~"));
}
```
(In any case it should indicate which separator was used.)

Also, maybe we should throw an exception instead of silently continue the encoding when "wrong" parent separator is used?